packages/core/upload/admin/src/components/AssetDialog/BrowseStep/index.js

```diff
@@ -231,99 +232,136 @@ export const BrowseStep = ({
-                            {
-                              id: getTrad('list.folder.subtitle'),
-                              defaultMessage:
-                                '{folderCount, plural, =0 {# folder} one {# folder} other {# folders}}, {filesCount, plural, =0 {# asset} one {# asset} other {# assets}}',
-                            },
-                            {
-                              folderCount: folder.children.count,
-                              filesCount: folder.files.count,
-                            }
-                          )}
-                        </TypographyMaxWidth>
-                      </Flex>
-                    </FolderCardBodyAction>
-                  </FolderCardBody>
-                </FolderCard>
-              </GridItem>
-            );
-          })}
-        </FolderGridList>
+          selected={selectedAssets}
+          shouldDisableBulkSelect={!multiple}
+          sortQuery={queryObject?.sort ?? ''}
+        />
       )}
 
-      {assetCount > 0 && folderCount > 0 && (
-        <Box paddingTop={6}>
-          <Divider />
-        </Box>
-      )}
+      {isGridView && (
+        <>
+          {folderCount > 0 && (
+            <FolderGridList
+              title={
+                (((isSearchingOrFiltering && assetCount > 0) || !isSearchingOrFiltering) &&
+                  formatMessage(
+                    {
+                      id: getTrad('list.folders.title'),
+                      defaultMessage: 'Folders ({count})',
+                    },
+                    { count: folderCount }
+                  )) ||
+                ''
+              }
+            >
+              {folders.map((folder) => {
+                return (
+                  <GridItem col={3} key={`folder-${folder.id}`}>
+                    <FolderCard
+                      ariaLabel={folder.name}
+                      id={`folder-${folder.id}`}
+                      onClick={() => handleClickFolderCard(folder.id)}
+                      cardActions={
+                        onEditFolder && (
+                          <IconButton
+                            icon={<PencilIcon />}
+                            aria-label={formatMessage({
+                              id: getTrad('list.folder.edit'),
+                              defaultMessage: 'Edit folder',
+                            })}
+                            onClick={() => onEditFolder(folder)}
+                          />
+                        )
+                      }
+                    >
+                      <FolderCardBody>
+                        <FolderCardBodyAction onClick={() => handleClickFolderCard(folder.id)}>
+                          <Flex as="h2" direction="column" alignItems="start" maxWidth="100%">
+                            <TypographyMaxWidth fontWeight="semiBold" ellipsis>
+                              {folder.name}
+                              <VisuallyHidden>:</VisuallyHidden>
```

Comment:
why is this visually hidden? 😅

Reply:
well good question aha
we don't want to see it in the design that's for sure
but why did we add it in the first place 🤔
maybe because it structures the voice reading to separate title and how many folders/assets 
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/71838159/203974126-d37d74b8-7fd9-4e6c-a8f3-45d31786546e.png">

Reply:
decided: change the ":" with "-"
comment [added](https://github.com/strapi/strapi/pull/14995/commits/61c8c05365720c08daadf3883524b6e19e0d399a) explaining the choice